mat/lu.go

```diff
@@ -22,27 +23,27 @@ type LU struct {
 	cond  float64
 }
 
-// updateCond updates the stored condition number of the matrix. Norm is the
-// norm of the original matrix. If norm is negative it will be estimated.
-func (lu *LU) updateCond(norm float64) {
+// updateCond updates the stored condition number of the matrix. anorm is the
+// norm of the original matrix. If anorm is negative it will be estimated.
+func (lu *LU) updateCond(anorm float64, norm lapack.MatrixNorm) {
 	n := lu.lu.mat.Cols
 	work := getFloats(4*n, false)
 	defer putFloats(work)
 	iwork := getInts(n, false)
 	defer putInts(iwork)
-	if norm < 0 {
+	if anorm < 0 {
 		// This is an approximation. By the definition of a norm, ||AB|| <= ||A|| ||B||.
 		// The condition number is ||A|| || A^-1||, so this will underestimate
```

Comment:
Again unrelated, but isn't it `overestimate`? `κ(A) := |A| |A^-1| = |LU| |A^-1| <= |L| |U| |A^-1|`